File: monitor-view/src/app/shared/entities/Composant.ts
import {ServerHote} from './ServerHote';

export interface IComposant {
  id?: number;
  nom?: string;
  serverHote?: ServerHote;
}

export class Composant implements IComposant{
  id: number;
  nom: string;
  serverHote: ServerHote;
  constructor(id: number, nom: string, serverHote: ServerHote) {
    this.id = id;
    this.nom = nom;
    this.serverHote = serverHote;
  }
}

File: monitor-data/src/main/java/africa/atps/monitordata/models/Composant.java
package africa.atps.monitordata.models;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;


@Entity
@AllArgsConstructor
@Getter
@Setter
@NoArgsConstructor
@Inheritance(strategy = InheritanceType.JOINED)
public class Composant implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String nom;

   /* @ManyToOne
    private ServerHote serverHote;*/


}
